packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -639,11 +642,47 @@ export function makeVirtualObjectManager(
   }
 
   function defineKind(tag, init, actualize, finish) {
-    return defineKindInternal(tag, init, actualize, finish, false);
+    const kindID = `${allocateExportID()}`;
+    return defineKindInternal(kindID, tag, init, actualize, finish, false);
+  }
+
+  let kindIDID;
+  const kindDescriptors = new WeakMap();
+
+  function reanimateDurableKindID(vobjID, _proforma) {
+    const { subid: kindID } = parseVatSlot(vobjID);
+    const raw = syscall.vatstoreGet(`vom.kind.${kindID}`);
+    assert(raw, X`unknown kind ID ${kindID}`);
+    const durableKindDescriptor = JSON.parse(raw);
+    const kindHandle = Far('kind', {});
+    kindDescriptors.set(kindHandle, durableKindDescriptor);
+    return kindHandle;
   }
 
-  function defineDurableKind(tag, init, actualize, finish) {
-    return defineKindInternal(tag, init, actualize, finish, true);
+  const makeKindHandle = tag => {
+    if (!kindIDID) {
+      kindIDID = `${allocateExportID()}`;
```

Comment:
I'm a bit bothered by the idea that this will differ from one vat to another, depending upon when that vat happens to make it's first handle.. I think that's going to make debugging harder. It overlaps with a similar concern about the IDs used for the standard virtual collections.

Is there any good way to make these values be the same for all vats?

Reply:
I don't think there's a _good_ way, though it's certainly quite doable.  The reason I did it the way it works now is that many vats don't need these things at all and so they only get created/allocated the first time somebody asks for one.  We could pre-allocate these for every vat, which would yield uniformity, but then we'd be carrying the overhead (mostly in the form of junk database entries, which I've found to be its own source of confusion when debugging) in all the places it's not used.  In practice I haven't found the variance of the IDs from one vat to another to be a debugging issue, since if you are looking at things at that low a level you see the record of the ID allocations along with their use.  And of course you typically debug in the context of a single vat and in that context the IDs stay the same from one debug run to another.